libevmasm/AssemblyStack.h

```diff
@@ -0,0 +1,58 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#pragma once
+
+#include <map>
+#include <string>
+
+#include <libsolutil/JSON.h>
+
+#include <libevmasm/Assembly.h>
+#include <libevmasm/LinkerObject.h>
+
+namespace solidity::evmasm
+{
+
+class AssemblyStack
```

Comment:
We should really call it `EVMAssemblyStack`. We used to have a class called `AssemblyStack` not so long ago and we just renamed it to `YulStack`. Every time I see this I think of that other class.
```suggestion
class EVMAssemblyStack
```

Reply:
Ok. Renaming it to `EvmAssemblyStack`.

Reply:
I just renamed it now to `EVMAssemblyStack`. But wouldn't `EvmAssemblyStack` be better readable? Because I used capitalised `EVM` also somewhere else here, I'm changing also this.